src/components/FilterButton/FilterButton.js

```diff
@@ -77,28 +84,39 @@ class FilterButton extends PureComponent {
         trackerLayer.setFilter(filterFc);
       } else {
         this.updatePermalink(true);
-        trackerLayer.setFilter(null);
+        const parameters = qs.parse(window.location.search.toLowerCase());
+        const lineParam = parameters[TrajservLayer.LINE_FILTER];
+        const routeParam = parameters[TrajservLayer.ROUTE_FILTER];
+        const opParam = parameters[TrajservLayer.OPERATOR_FILTER];
+        const { regexPublishedLineName } = trackerLayer.options;
+        filterFc = null;
+        if (lineParam || routeParam || opParam || regexPublishedLineName) {
+          filterFc = TrajservLayer.createFilter(
+            lineParam ? lineParam.split(',') : undefined,
+            routeParam ? routeParam.split(',') : undefined,
+            opParam ? opParam.split(',') : undefined,
+            regexPublishedLineName,
+            false,
+          );
+        }
+
+        trackerLayer.setFilter(filterFc);
```

Comment:
seems a lot similar to addTrackerFilters function, is there a possibility to reuse it.